packages/design/jobberStyle.js

```diff
@@ -154,41 +154,41 @@ function getResolvedCSSVars(cssProperties) {
 
 function getResolvedSCSSVariables(cssProperties) {
   const allKeys = Object.keys(cssProperties);
-  const sizeVariables = ["border", "radius"];
-  const simpleVariables = [
-    "color",
-    "timing",
-    "elevation",
-    "lineHeight",
-    "fontFamily",
-    "letterSpacing-base",
-  ];
-  const calcVariables = ["space", "letterSpacing-loose", "fontSize"];
 
   return allKeys.reduce((acc, cssVar) => {
-    const isSizeVariable = sizeVariables.some(v => cssVar.includes(v));
-    const isSimpleVariable = simpleVariables.some(v => cssVar.includes(v));
-    const isCalcVariable = calcVariables.some(v => cssVar.includes(v));
-
-    if (isSimpleVariable) {
-      return [...acc, `$${cssVar}: ${resolvedCssVars[cssVar]};`];
-    } else if (isCalcVariable) {
-      const calcRegexResult = regexExpressions.calculations.exec(
-        customProperties["--" + cssVar],
-      );
-      return [...acc, `$${cssVar}: ${handleCalc(calcRegexResult)}px;`];
-    } else if (isSizeVariable) {
-      const suffix = customProperties["--" + cssVar].includes("%") ? "%" : "px";
-      return [...acc, `$${cssVar}: ${resolvedCssVars[cssVar]}${suffix};`];
-    } else if (cssVar.includes("shadow")) {
-      const resolvedShadowProperty = resolveShadow(
-        customProperties["--" + cssVar],
-      );
-
-      return [...acc, `$${cssVar}: ${resolvedShadowProperty};`];
-    } else {
-      return acc;
+    const customPropertyValue = customProperties["--" + cssVar];
+    const variableType = getVariableType(cssVar);
+    let propertyValue = "";
+
+    switch (variableType) {
```

Comment:
thoughts on breaking this `switch` out into its own function? Separate those concerns of reducing and figuring out what value?

```js
const propertyValue = getPropertyValue();
if (propertyValue) {
  return [...acc, `$${cssVar}: ${propertyValue};`];
}

return acc;
```

IMO, it makes the reducer simpler and straight to the point. if there's a property value, add it; otherwise, don't. That way you won't have to reassign a variable. Up to you tho

Reply:
I liked this idea, I pushed the changes related to your comment. Thanks!